### PR TITLE
Implemented endpoint to refresh rss feeds, written tests for Tribler.Core.Modules.channel package

### PR DESCRIPTION
**Pull request by devos50**

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis ready for your review.

File: Tribler/Core/Modules/channel/channel.py [hunk 1]
```diff
@@ -39,6 +39,9 @@ def name(self):
     def get_rss_feed_url_list(self):
         return [url for url in self._rss_feed_dict.iterkeys()]
 
+    def refresh_all_feeds(self):
```
Comment (lfdversluis) [inline Tribler/Core/Modules/channel/channel.py]: might want to let the button in the wx gui call this as well :p

File: Tribler/Test/Core/base_test.py [hunk 2]
```diff
@@ -1,5 +1,12 @@
 from Tribler.Test.test_as_server import AbstractServer
 
 
+class MockObject(object):
```
Comment (lfdversluis) [inline Tribler/Test/Core/base_test.py]: You (and I) may had to introduce this earlier, we now make custom objects in tests already.

File: doc/Tribler REST API.md [hunk 3]
```diff
@@ -179,6 +180,10 @@ Add a RSS feed to your channel. Returns error 409 (Conflict) if the supplied RSS
 
 Delete a RSS feed from your channel. Returns error 404 if the RSS feed that is being removed does not exist. Note that the rss feed url should be URL-encoded.
 
+## `POST /mychannel/recheckfeeds`
+
+Recheck all rss feeds in your channel. Returns error 404 if you channel does not exist.
```
Comment (lfdversluis) [inline doc/Tribler REST API.md]: Rechecks ?

**Comment by lfdversluis** [comment]
Comments given

File: Tribler/Core/Modules/channel/channel.py (hunk 1 above)
Comment (devos50) [inline Tribler/Core/Modules/channel/channel.py]: Good point, done! The button now does something useful lol

File: Tribler/Test/Core/base_test.py (hunk 2 above)
Comment (devos50) [inline Tribler/Test/Core/base_test.py]: Yeah I agree, I got a bit tired of defining separate classes every time and last week, @whirm told me about creating classes that derive from `object` which is an easy way to mock objects so I decided to go with this approach.

**Comment by devos50** [comment]
@whirm processed feedback. I've tested the recheck feeds button in the GUI manually. Can you take a look?

File: Tribler/Main/vwxGUI/channel.py [hunk 4]
```diff
@@ -1536,6 +1536,9 @@ def OnDeleteRss(self, event):
         my_channel_object.remove_rss_feed(item.url)
 
     def OnRefreshRss(self, event):
+        my_channel_object = self.guiutility.utility.session.lm.channel_manager.get_my_channel(self.channel.id)
+        my_channel_object.refresh_all_feeds()
```
Comment (whirm) [inline Tribler/Main/vwxGUI/channel.py]: Is that a blocking call?

Comment (devos50) [inline Tribler/Main/vwxGUI/channel.py]: iirc, fetching the RSS feeds and torrents happens asynchronously using Twisted.

**Comment by whirm** [comment]
Ok, squash it then!

**Comment by devos50** [comment]
```
test_ffsearch (Tribler.Test.test_remote_search.TestMixedRemoteSearch) ... Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@3/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@3/tribler/Tribler/Core/Modules/tracker_manager.py", line 98, in update_tracker_info
    tracker_info = self._tracker_dict[tracker_url]
exceptions.TypeError: 'NoneType' object has no attribute '__getitem__'
```

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
Same error as above

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm ready for merge

**Comment by whirm** [comment]
Needs rebase

**Comment by devos50** [comment]
Remote torrent search test:

```
Test raised 1 unhandled Twisted exceptions, last one was: Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@3/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux@3/tribler/Tribler/Core/Modules/tracker_manager.py", line 98, in update_tracker_info
    tracker_info = self._tracker_dict[tracker_url]
exceptions.TypeError: 'NoneType' object has no attribute '__getitem__'
```

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
```
Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/endpoint.py", line 250, in dispersythread_data_came_in
    u"standalone_ep")
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/dispersy/dispersy.py", line 1464, in on_incoming_packets
    (str(packet[0]), repr(packet[1])) for packet in packets]
exceptions.AssertionError: [('<Candidate 178.209.103.30:12847>', "'A\\x00\\xbf\\x9d(\\x8f\\x82v\\x00\\x00\\x00\\x00\\x00\\x10\\x00\\x00\\xda4\\x00\\x00'")]
```

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm ready for merge

**Pull request merged**